src/tls/openssl.c

```diff
@@ -133,14 +133,60 @@ static void tls_context_destroy(void *ctx_backend)
     flb_free(ctx);
 }
 
+#ifdef _MSC_VER
+static int windows_load_system_certificates(struct tls_context *ctx)
+{
+    int ret;
+
+    HANDLE h;
+    PCCERT_CONTEXT cert = NULL;
+    X509_STORE *store = SSL_CTX_get_cert_store(ctx->ctx);
+    X509 *x509;
+
+    h = CertOpenSystemStoreA(0, "Root");
+    if (h == NULL) {
+        flb_error("[tls] Cannot open cert store: %i", GetLastError());
+        return -1;
+    }
+
+    while (cert = CertEnumCertificatesInStore(h, cert)) {
+        if (cert->dwCertEncodingType & X509_ASN_ENCODING) {
+            // Decode the certificate into X509 struct
+            // `buf` is necessary per openssl docs because the pointer is incremented
+            const unsigned char *buf = cert->pbCertEncoded;
+            x509 = d2i_X509(NULL, &buf, cert->cbCertEncoded);
+            if (!x509) {
+                flb_debug("[tls] Cannot parse a certificate. skipping...");
+                continue;
+            }
+
+            // Add X509 struct to the openssl cert store
+            ret = X509_STORE_add_cert(store, x509);
```

Comment:
I assume this is well tested but while checking the man page for this call to verify what happens to x509 I saw this : 

```
X509_STORE_add_cert() and X509_STORE_add_crl() add the respective object to the X509_STORE's local storage. Untrusted objects should not be added in this way. The added object's reference count is incremented by one, hence the caller retains ownership of the object and needs to free it when it is no longer needed.
```

I might be wrong but I interpreted that as meaning that they don't make a copy and releasing our instance before the validation is performed would probably end in an illegal access. Could you clarify this for me?

https://www.openssl.org/docs/man1.1.1/man3/X509_STORE_add_cert.html